pyface/i_heading_text.py

```diff
@@ -10,32 +10,109 @@
 
 """ Heading text. """
 
+import warnings
 
-from traits.api import HasTraits, Instance, Int, Interface, Str
-
-
-from pyface.i_image_resource import IImageResource
+from traits.api import HasTraits, Int, Interface, Str, observe
 
 
 class IHeadingText(Interface):
-    """ Heading text. """
+    """ A widget which shows heading text in a panel. """
 
     # 'IHeadingText' interface ---------------------------------------------
 
-    #: Heading level.
-    #
-    # fixme: Currently we ignore anything but one, but in future we could
-    # have different visualizations based on the level.
+    #: Heading level.  This is currently unused.
     level = Int(1)
 
     #: The heading text.
     text = Str("Default")
 
-    #: The background image.
-    image = Instance(IImageResource)
-
 
 class MHeadingText(HasTraits):
     """ The mixin class that contains common code for toolkit specific
     implementations of the IHeadingText interface.
     """
+
+    # 'IHeadingText' interface ---------------------------------------------
+
+    #: Heading level.  This is currently unused.
+    level = Int(1)
+
+    #: The heading text.
+    text = Str("Default")
+
+    def __init__(self, parent=None, **traits):
+        """ Creates the heading text. """
+
+        if "image" in traits:
+            warnings.warn(
+                "background images are no-longer supported for Wx and the "
+                "'image' trait will be removed in a future Pyface update",
+                PendingDeprecationWarning,
+            )
+
+        create = traits.pop("create", True)
+
+        # Base class constructor.
+        super().__init__(parent=parent, **traits)
+
+        if create:
+            # Create the widget's toolkit-specific control.
+            self.create()
+            warnings.warn(
+                "automatic widget creation is deprecated and will be removed "
+                "in a future Pyface version, use create=False and explicitly "
+                "call create() for future behaviour",
+                PendingDeprecationWarning,
+            )
```

Comment:
same comment here regd an issue - and we can put this in the 8.0.0 milestone too if that makes sense.

Reply:
This is part of #729 (particularly the comment at the bottom about automatic creation in `__init__`).

But I don't want to add it to a milestone until all of them are dealt with (ie. if I don't get it done this week it might bet Pyface 9 where we finally get rid of them).

Reply:
I've added a comment to #729 which goes over the idea for the sequence of events needed to change the API safely.